test/metabase/upload_test.clj

```diff
@@ -558,3 +562,35 @@
             (testing "Check the data was uploaded into the table correctly"
               (is (= [["Malcolm"] ["\\."] ["Han"]]
                      (rows-for-table table))))))))))
+
+(deftest mysql-settings-test
+  (testing "Ensure that local_infile is set to true for better MySQL testing"
+    (mt/test-drivers [:mysql]
+      (is (= "ON" (-> (sql-jdbc.conn/db->pooled-connection-spec (mt/db))
+                      (jdbc/query
+                       ["show global variables like 'local_infile'"])
+                      first
+                      :value))))))
+
+(deftest load-from-csv-mysql-slow-way-test
+  (testing "MySQL upload should work fine with local_infile disabled"
+    (mt/test-drivers [:mysql]
+      (mt/with-empty-db
+        (with-redefs [mysql/get-global-variable (constantly "OFF")]
+          (upload/load-from-csv
+           driver/*driver*
+           (mt/id)
+           "upload_test"
+           (csv-file-with ["id,ship,captain"
+                           "1,Serenity,Malcolm Reynolds"
+                           "2,Millennium Falcon,Han Solo"])))
+        (testing "Table and Fields exist after sync"
+          (sync/sync-database! (mt/db))
+          (let [table (t2/select-one Table :db_id (mt/id))]
+            (is (=? {:name #"(?i)upload_test"} table))
+            (testing "Check the data was uploaded into the table correctly"
+              (is (= ["id", "ship", "captain"]
+                     (column-names-for-table table)))
+              (is (= [[1 "Serenity" "Malcolm Reynolds"]
+                      [2 "Millennium Falcon" "Han Solo"]]
+                     (rows-for-table table))))))))))
```

Comment:
Unintentionally discovered that (due to quoting) the way I was calling `(driver/insert-into :sql-jdbc ...)` was no good, so I wrote a test for it.